### PR TITLE
docs: Developer guide improvements

### PR DESCRIPTION
**Pull request by ivarconr**
## About the changes

I played with the OpenAI APIs to see how easy it would be to get help improving the docs, and this is the results of getting it to improve the developer guide.

### CODE REVIEW

**Review by thomasheartman — APPROVED** [review]

File: website/docs/contributing/developer-guide.md [hunk 1]
```diff
@@ -4,25 +4,19 @@ title: Developer guide
 
 ## Introduction {#introduction}
 
-This repository contains two main parts. The backend and the frontend of unleash. The backend is a Node.js application that is built using TypeScript. The frontend is a React application that is built using TypeScript. The backend specific code can be found in the `src` lib folder. The frontend specific code can be found in the `frontend` folder.
+This repository consists of two main parts: the backend and frontend of Unleash. The backend is a Node.js application built with TypeScript, while the frontend is a React application also built with TypeScript. You can find code specific to the backend in the `src` lib folder and code specific to the frontend in the `frontend` folder.
```
Comment (ivarconr) [inline website/docs/contributing/developer-guide.md:7]: I like this improvement, makes it easier to read.

File: website/docs/contributing/developer-guide.md [hunk 2]
```diff
@@ -4,25 +4,19 @@ title: Developer guide
...
 
 ## Development philosophy
 
-The development philosophy at unleash is centered at delivering high quality software. We do this by following a set of principles that we believe will help us achieve this goal. We believe that these principles will also help us deliver software that is easy to maintain and extend, serving as our north star.
+The development philosophy at Unleash is centered on delivering high-quality software. We achieve this by following a set of principles that we believe will also make the software easy to maintain and extend, serving as our guide.
```
Comment (ivarconr) [inline website/docs/contributing/developer-guide.md:11]: +1

File: website/docs/contributing/developer-guide.md [hunk 3]
```diff
@@ -4,25 +4,19 @@ title: Developer guide
...
 
-We believe that the following principles will help us achieve our goal of delivering high quality software:
+We believe that the following principles are essential in achieving our goal:
 
-* We test our code always
+- Testing code always: Software engineering is challenging, and it's crucial to take every precaution necessary to avoid introducing bugs. Thus, testing is an effective way to accomplish this aim. We prefer automation over manual testing.
+ 
+- Writing code that is easy to understand and maintain: Code serves as language – a means of communicating intent with readers efficiently. Therefore, writing clean code matters significantly for us because it keeps our codebase manageable while maintaining speed in the long run.
```
Comment (ivarconr) [inline website/docs/contributing/developer-guide.md:17]: Unsure if this rewrite is better. Thoughts?

File: website/docs/contributing/developer-guide.md [hunk 4]
```diff
@@ -4,25 +4,19 @@ title: Developer guide
...
-
-We believe code is a language. Written code is a way to communicate intent. It's about explaining to the reader what this code does, in the shortest amount of time possible. As such, writing clean code is supremely important to us. We believe that this contributes to keeping our codebase maintainable, and helps us maintain speed in the long run.
-
-* We think about solutions before comitting
-
-We don't jump to implementation immediately. We think about the problem at hand, and try to examine the impact that this solution may have in a multitude of scenarios. As our product core is open source, we need to balance the solutions and avoid implementations that may be cumbersome for our community. The need to improve our paid offering must never come at the cost of our open source offering.
+- Considering solutions before committing: Rather than jumping straight into implementation, we examine how various scenarios may affect a particular solution concerning the problem at hand's impact. As our product core is open source-oriented, balancing solutions without burdening communities becomes vital; improving paid offerings should never come at Open-source offering cost.
```
Comment (ivarconr) [inline website/docs/contributing/developer-guide.md:19]: I'm not sure about this rewrite.

File: website/docs/contributing/developer-guide.md (hunk 3 above)
Comment (thomasheartman) [reply website/docs/contributing/developer-guide.md:17]: I'm fine with either. I like the old one, but I think this one works well too.

File: website/docs/contributing/developer-guide.md (hunk 4 above)
Comment (thomasheartman) [reply website/docs/contributing/developer-guide.md:19]: Eh, I think the old one is better here. There's some weird/awkward phrasings here ("our product core is open source-oriented", "concerning the problem at hand's impact", "paid offerings should never come at Open-source offering cost") and a weird capitalization of "Open-source".

File: website/docs/contributing/developer-guide.md (hunk 1 above)
Comment (sighphyre) [reply website/docs/contributing/developer-guide.md:7]: +1

File: website/docs/contributing/developer-guide.md (hunk 3 above)
Comment (sighphyre) [reply website/docs/contributing/developer-guide.md:17]: Completely neutral on this one. I don't think it's worse but it's consistent with the tone of other parts that have been rewritten

File: website/docs/contributing/developer-guide.md (hunk 4 above)
Comment (sighphyre) [reply website/docs/contributing/developer-guide.md:19]: I think it's overall simpler but this sentence here:

> The need to improve our paid offering must never come at the cost of our open source offering.

Feels very awkward to me

What bothers you about this one?

File: website/docs/contributing/developer-guide.md (hunk 3 above)
Comment (FredrikOseberg) [reply website/docs/contributing/developer-guide.md:17]: I think the old one wins here. The old version has more heart while this feels like just another generic AI summary.

File: website/docs/contributing/developer-guide.md (hunk 4 above)
Comment (FredrikOseberg) [reply website/docs/contributing/developer-guide.md:19]: I have the same comment as above for this one: I think the old one wins here. The old version has more heart while this feels like just another generic AI summary.

File: website/docs/contributing/developer-guide.md (hunk 2 above)
Comment (FredrikOseberg) [reply website/docs/contributing/developer-guide.md:11]: +1

**Pull request merged**